### PR TITLE
Added API to fetch RNG state.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

**Comment by dlibenzi** [comment]
Fixes #2093

**Comment by myleott** [comment]
Will you also add a set_rng_state? It seems from the docstring that you return some kind of running seed. Is the idea that you use `set_rng_seed(get_rng_state())`? Note that for many RNG the state is not just a single int. Rather, the initial state is based on the seed, but then changes as the RNG is called.

The main functionality I care about is that you can set an initial seed, generate K times, save the state, generate a few more, and finally restore the state to exactly where it was after the initial K generations. Something like:

```python
xm.set_rng_seed(0)

# first call
torch.zeros(5, device=xm.xla_device()).uniform_()
# tensor([0.8690, 0.0799, 0.6938, 0.8282, 0.0444], device='xla:1')

rng_state = xm.get_rng_state()

# second call
expected = torch.zeros(5, device=xm.xla_device()).uniform_()
# tensor([0.9650, 0.4818, 0.2164, 0.2308, 0.8543], device='xla:1')

xm.set_rng_state(rng_state)

# same as original second call
actual = torch.zeros(5, device=xm.xla_device()).uniform_()
# tensor([0.9650, 0.4818, 0.2164, 0.2308, 0.8543], device='xla:1')

assert torch.all(expected == actual)
```

**Comment by dlibenzi** [comment]
There is no live complex state on XLA. Just a 64bit seed.
So this is OK:
```Python
set_seed(get_state())
```

I was debating on how to name it, as usually the seed is not a moving entity, while the state is.
So calling it a get-seed thing did not sound nice.
But OTOH setting the seed with a state does not sound too good either.
An alternative is to rename set-seed as set-state thing.

**Comment by dlibenzi** [comment]
> There is no live complex state on XLA. Just a 64bit seed.
> So this is OK:
> 
> ```python
> set_seed(get_state())
> ```
> 
> I was debating on how to name it, as usually the seed is not a moving entity, while the state is.
> So calling it a get-seed thing did not sound nice.
> But OTOH setting the seed with a state does not sound too good either.
> An alternative is to rename set-seed as set-state thing.

I opted for the latter, to unify naming.
